aiida/storage/psql_dos/backend.py

```diff
@@ -453,3 +450,96 @@ def get_info(self, detailed: bool = False) -> dict:
+        container = Container(str(get_filepath_container(self.profile)))
+
+        # check that the AiiDA profile is not locked and request access for the duration of this backup process
+        # (locked means that possibly a maintenance operation is running that could interfere with the backup)
+        try:
+            ProfileAccessManager(self._profile).request_access()
+        except LockedProfileError:
+            raise backup_utils.BackupError('The profile is locked!')
+
+        # step 1: first run the storage maintenance version that can safely be performed while aiida is running
+        self.maintain(full=False, compress=True)
+
+        # step 2: dump the PostgreSQL database into a temporary directory
+        pg_dump_exe = manager.exes['pg_dump']
+        with tempfile.TemporaryDirectory() as temp_dir_name:
+            psql_temp_loc = pathlib.Path(temp_dir_name) / 'db.psql'
+
+            env = os.environ.copy()
+            env['PGPASSWORD'] = cfg['database_password']
+            cmd = [
+                pg_dump_exe, f'--host={cfg["database_hostname"]}', f'--port={cfg["database_port"]}',
+                f'--dbname={cfg["database_name"]}', f'--username={cfg["database_username"]}', '--no-password',
+                '--format=p', f'--file={str(psql_temp_loc)}'
+            ]
+            try:
+                subprocess.run(cmd, check=True, env=env)
+            except subprocess.CalledProcessError as exc:
+                raise backup_utils.BackupError(f'pg_dump: {exc}')
+
+            if psql_temp_loc.is_file():
+                STORAGE_LOGGER.info(f'Dumped the PostgreSQL database to {str(psql_temp_loc)}')
+            else:
+                raise backup_utils.BackupError(f"'{str(psql_temp_loc)}' was not created.")
+
+            # step 3: transfer the PostgreSQL database file
+            manager.call_rsync(psql_temp_loc, path, link_dest=prev_backup, dest_trailing_slash=True)
+
+        # step 4: back up the disk-objectstore
+        backup_utils.backup_container(
+            manager, container, path / 'container', prev_backup=prev_backup / 'container' if prev_backup else None
+        )
+
+        # step 5: back up aiida config.json file
+        try:
+            config = get_config()
+            manager.call_rsync(pathlib.Path(config.filepath), path)
+        except (exceptions.MissingConfigurationError, exceptions.ConfigurationError):
+            STORAGE_LOGGER.warning('aiida config.json not found!')
+
+    def backup(
+        self,
+        dest: str,
+        keep: int,
+        exes: dict,
+    ) -> bool:
+
+        try:
+            backup_manager = backup_utils.BackupManager(dest, STORAGE_LOGGER, exes=exes, keep=keep)
+            backup_manager.backup_auto_folders(lambda path, prev: self._backup(backup_manager, path, prev))
+        except backup_utils.BackupError as exc:
+            STORAGE_LOGGER.error(f'Error: {exc}')
+            return False
+
+        return True
```

Comment:
This should be converted to `StorageBackupError`. We should create it in `aiida.common.exceptions`.